[3/3] 3D Composability - move tp dp tests

mori360 opened the pull request:
pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py


Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #129802
* #129801



cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @tianyu-l @chauhang

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update on "[3/3] 3D Composability - move tp dp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py





cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse tianyu-l chauhang

[ghstack-poisoned]

XilunWu commented on .ci/pytorch/multigpu-test.sh:
```diff
@@ -44,8 +44,6 @@ time python test/run_test.py --verbose -i distributed/_tensor/test_dtensor_compi
 # DTensor/TP tests
-time python test/run_test.py --verbose -i distributed/tensor/parallel/test_ddp_2d_parallel
-time python test/run_test.py --verbose -i distributed/tensor/parallel/test_fsdp_2d_parallel
 time python test/run_test.py --verbose -i distributed/tensor/parallel/test_tp_examples
```
you will also need to add the new test file here

mori360 replied:
> you will also need to add the new test file here

new tests are included in [test_2d_composability](https://github.com/pytorch/pytorch/pull/129802/files#diff-34b5ad4462d9fccf96dcd7c22fe3f3eaa7c23cb60886281be02f00d32b5d4f4aR54)

XilunWu reviewed (commented):
And we need to address this change in other files besides `multigpu-test.sh`

mori360 committed:
Update on "[3/3] 3D Composability - move tp dp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py





cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse tianyu-l chauhang

[ghstack-poisoned]

fduwjj commented on test/distributed/_composable/test_composability/test_2d_composability.py:
```diff
@@ -2,30 +2,136 @@
+    return model, twod_model, dp_pg
+
+
+class SimpleModel(nn.Module):
```
Not need to do in this PR, but we had so many duplicate codes in the past, it would be great if we can refactor them to consolidate them a little bit.

mori360 committed:
Update on "[3/3] 3D Composability - move tp dp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py





cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse tianyu-l chauhang

[ghstack-poisoned]

fduwjj commented:
Looks like there are still lint error and test failure? Also looks like periodic / linux-focal-rocm6.1-py3.8 / test (distributed, 2, 2, linux.rocm.gpu) (push) timed out. Do we see this in the past?

mori360 commented:
> Looks like there are still lint error and test failure? Also looks like periodic / linux-focal-rocm6.1-py3.8 / test (distributed, 2, 2, linux.rocm.gpu) (push) timed out. Do we see this in the past?

The lint error comes with `test/export/test_converter.py`, which is not touched in this PR.
Will look into the `periodic / linux-focal-rocm6.1-py3.8 / test (distributed, 2, 2, linux.rocm.gpu) (push) timed out`

mori360 committed:
Update on "[3/3] 3D Composability - move tp dp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py





cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse tianyu-l chauhang

[ghstack-poisoned]

mori360 committed:
Update on "[3/3] 3D Composability - move tp dp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py





cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse tianyu-l chauhang

[ghstack-poisoned]

mori360 committed:
Update on "[3/3] 3D Composability - move tp dp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py





cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse tianyu-l chauhang

[ghstack-poisoned]

mori360 committed:
Update on "[3/3] 3D Composability - move tp dp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py





cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse tianyu-l chauhang

[ghstack-poisoned]

mori360 committed:
Update on "[3/3] 3D Composability - move tp dp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py





cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse tianyu-l chauhang

[ghstack-poisoned]

mori360 committed:
Update on "[3/3] 3D Composability - move tp dp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py





cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse tianyu-l chauhang

[ghstack-poisoned]

fduwjj reviewed (approved):
LGTM

mori360 commented:
@pytorchbot merge

awgu commented on test/distributed/_composable/test_composability/test_2d_composability.py:
```diff
@@ -2,30 +2,131 @@
-from torch.distributed._tensor import init_device_mesh
+from torch.distributed._tensor import (
+    DTensor,
+    DTensor as DT,
```
Can we unify under `DTensor` and get rid of `DT`? (in a follow-up PR)

awgu commented on test/distributed/_composable/test_composability/test_2d_composability.py:
```diff
@@ -2,30 +2,131 @@
 from torch.testing._internal.distributed.checkpoint_utils import with_temp_dir
 
 
+def init_model(device_type, model_parallel_size=2):
```
I agree that this needs some cleanup. These probably should not be defined at global scope but rather within the test class that uses them (or something like that).

mori360 replied:
Moved into the corresponding class

awgu commented on test/distributed/_composable/test_composability/test_2d_composability.py:
```diff
@@ -279,5 +380,419 @@ def parallelize(_model: Transformer, mesh: DeviceMesh, use_seq_parallel: bool):
         self.assertEqual(loss_no_cp2, loss_cp2)
 
 
+class Test2dParallelIntegration(DTensorTestBase):
```
I think this should be named in a way that makes it clear that this is testing FSDP1.

mori360 replied:
Renamed

awgu commented on test/distributed/_composable/test_composability/test_2d_composability.py:
```diff
@@ -279,5 +380,419 @@ def parallelize(_model: Transformer, mesh: DeviceMesh, use_seq_parallel: bool):
 
 
+class Test2dParallelIntegration(DTensorTestBase):
+    global LR
+    LR = 3e-5
```
not from this PR but this seems... strange to me
Why use a global variable and then set it inside the test class?

mori360 replied:
I make it the same style as the others (lr=...) in the latest commit

awgu reviewed (commented):
I understand that this is just code motion, but I think code motion is a good time to clean up things.

(It would probably also help build familiarity.)

cc: @weifengpy @fduwjj @wconstab @mori360 should we do some clean up in addition to just code motion?

Otherwise, it is not immediately clear to me the value of only moving things.

mori360 committed:
Update on "[3/3] 3D Composability - move tp dp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py





cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse tianyu-l chauhang

[ghstack-poisoned]

mori360 commented:
@pytorchbot merge

mori360 commented:
@pytorchbot merge

atalman commented:
@pytorchmergebot revert -c nosignal -m "Need to revert https://github.com/pytorch/pytorch/pull/129801 that got remerged"

mori360 committed:
Update on "[3/3] 3D Composability - move tp dp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py





cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse tianyu-l chauhang

[ghstack-poisoned]

mori360 committed:
Update on "[3/3] 3D Composability - move tp dp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
**DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py**
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py





cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse tianyu-l chauhang

[ghstack-poisoned]

mori360 commented:
@pytorchbot merge
